make/compiler/Makefile.gnu

```diff
@@ -162,10 +162,11 @@ GEN_CFLAGS += $(C_STD)
 # of Clang header files.
 #
 WARN_COMMONFLAGS = -Wall -Werror -Wpointer-arith -Wwrite-strings -Wno-strict-aliasing
-WARN_CXXFLAGS = $(WARN_COMMONFLAGS) -Wno-comment
+WARN_CXXFLAGS = $(WARN_COMMONFLAGS) -Wno-comment -Werror=zero-as-null-pointer-constant
```

Comment:
Will this flag be supported by all versions of g++ that we test against, or do we need to restrict it to certain versions?

Reply:
> Will this flag be supported by all versions of g++ that we test against, or do we need to restrict it to certain versions?

That's a good question.  GCC added in version 4.7. Clang has it in version 5.0. Do we support older versions?

Reply:
I'm not the best person to ask about that (and would take it to the chapel slack channel).  I don't believe we use gcc < 4.7.  I'm less certain about clang.